### DISCUSSION TITLE
How can i get pressed key in plugin

### DISCUSSION THREAD

**Discussion by crys1337** [SmartlyDressedGames/Unturned-3.x-Community#3944] [Q&A] [answered]
can u help me?

**Comment by GazziFX** [comment, marked as answer]
You can track keys bound as Plugins/Mods keys with callback
```csharp
PlayerInput.onPluginKeyTick += onPluginKeyTick;
```
and some other keys like movement or lean with
```csharp
player.input.keys[index]
```
P.S. It is recommended to read keys array from onPluginKeyTick to be in sync with inputs, since they can be received with variable delay

**Reply by crys1337** [reply]
how can i get key codes?

**Reply by GazziFX** [reply]
```csharp
keys[0] = base.player.movement.jump; 
keys[1] = false; 
keys[2] = false; 
keys[3] = base.player.stance.crouch; 
keys[4] = base.player.stance.prone; 
keys[5] = base.player.stance.sprint; 
keys[6] = base.player.animator.leanLeft; 
keys[7] = base.player.animator.leanRight; 
keys[8] = false; 
keys[9] = base.player.stance.localWantsToSteadyAim; 
```

**Comment by DanielWillett** [comment]
Side note, this is called 5 times per fixed update which may cause unintended behavior. If you want to just check once check if the key argument == 0 before running your logic.
